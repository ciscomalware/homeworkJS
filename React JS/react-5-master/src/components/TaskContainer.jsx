import React from 'react';
import { uniqueId } from 'lodash';
import TaskForm from './TaskForm';
import TaskList from './TaskList';

export default class TaskContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      titleValue: '',
      bodyValue: '',
      taskList: [],
    };
  }

  changeTitle = (e) => {
    e.preventDefault();
    this.setState({
      titleValue: e.target.value,
    });
  }

  changeBody = (e) => {
    e.preventDefault();
    this.setState({
      bodyValue: e.target.value,
    });
  }

  onSubmit = (e) => {
    e.preventDefault();
    this.setState({
      taskList: [
        {
          id: uniqueId(), title: this.state.titleValue, body: this.state.bodyValue, state: 'finish',
        },
        ...this.state.taskList,
      ],
      titleValue: '',
      bodyValue: '',
    });
  }

  finishTask = (id) => () => {
    this.setState({
      taskList: this.state.taskList.map((task) => (task.id === id
        ? { ...task, state: 'delete' }
        : task)),
    });
  }

  deleteTask = (id) => () => {
    this.setState({
      taskList: this.state.taskList.filter((task) => task.id !== id),
    });
  }

  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-12">
            <TaskForm
              title={this.state.titleValue}
              body={this.state.bodyValue}
              changeTitle={this.changeTitle}
              changeBody={this.changeBody}
              submit={this.onSubmit}
            />
          </div>
          <div className="col-12">
            <TaskList
              tasks={this.state.taskList}
              finish={this.finishTask}
              delete={this.deleteTask}
            />
          </div>
        </div>
      </div>
    );
  }
}
