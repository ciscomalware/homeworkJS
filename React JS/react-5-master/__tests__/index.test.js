import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount, } from 'enzyme';

import TaskContainer from '../src/components/TaskContainer';


configure({ adapter: new Adapter() });

test('TaskContainer - should work adding new task', () => {
  const component = mount(
    <TaskContainer />,
  );

  expect(component.html()).toMatchSnapshot();
  component.find('input').first().simulate('change', { target: { value: 'Hello, World!' } });
  component.find('input').last().simulate('change', { target: { value: 'This is my first task!' } });
  component.find('form').simulate('submit');
  expect(component.html()).toMatchSnapshot();
  component.find('input').first().simulate('change', { target: { value: 'Second Task!!' } });
  component.find('input').last().simulate('change', { target: { value: 'This is my seconds task!' } });
  component.find('form').simulate('submit');
  expect(component.html()).toMatchSnapshot();
});

test('TaskContainer - should work finishing task', () => {
  const component = mount(
    <TaskContainer />,
  );

  component.find('input').first().simulate('change', { target: { value: 'Hello, World!' } });
  component.find('input').last().simulate('change', { target: { value: 'This is my first task!' } });
  component.find('form').simulate('submit');
  component.find('input').first().simulate('change', { target: { value: 'Second Task!!' } });
  component.find('input').last().simulate('change', { target: { value: 'This is my seconds task!' } });
  component.find('form').simulate('submit');
  component.find('.card').first().find('button').simulate('click');
  expect(component.html()).toMatchSnapshot();
  component.find('.card').last().find('button').simulate('click');
  expect(component.html()).toMatchSnapshot();
});

test('TaskContainer - should work removing task', () => {
  const component = mount(
    <TaskContainer />,
  );

  component.find('input').first().simulate('change', { target: { value: 'Hello, World!' } });
  component.find('input').last().simulate('change', { target: { value: 'This is my first task!' } });
  component.find('form').simulate('submit');
  component.find('input').first().simulate('change', { target: { value: 'Second Task!!' } });
  component.find('input').last().simulate('change', { target: { value: 'This is my seconds task!' } });
  component.find('form').simulate('submit');
  component.find('.card').first().find('button').simulate('click');
  component.find('.card').first().find('button').simulate('click');
  expect(component.html()).toMatchSnapshot();
  component.find('.card').last().find('button').simulate('click');
  component.find('.card').last().find('button').simulate('click');
  expect(component.html()).toMatchSnapshot();
});