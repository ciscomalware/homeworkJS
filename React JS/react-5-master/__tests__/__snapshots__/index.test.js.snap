// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskContainer - should work adding new task 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Заголовок задачи...\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Текст задачи...\\" value=\\"\\"></div><button class=\\"btn btn-outline-secondary\\" type=\\"submit\\">Добавить задачу</button></form></div><div class=\\"col-12\\"><div class=\\"row\\"></div></div></div></div>"`;

exports[`TaskContainer - should work adding new task 2`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Заголовок задачи...\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Текст задачи...\\" value=\\"\\"></div><button class=\\"btn btn-outline-secondary\\" type=\\"submit\\">Добавить задачу</button></form></div><div class=\\"col-12\\"><div class=\\"row\\"><div class=\\"col-3\\"><div class=\\"card m-2 text-white bg-primary\\"><div class=\\"card-header\\"><h3 class=\\"card-title\\">Hello, World!</h3></div><div class=\\"card-body\\"><p class=\\"card-text\\">This is my first task!</p></div><div class=\\"card-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-light\\">Завершить задачу</button></div></div></div></div></div></div></div>"`;

exports[`TaskContainer - should work adding new task 3`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Заголовок задачи...\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Текст задачи...\\" value=\\"\\"></div><button class=\\"btn btn-outline-secondary\\" type=\\"submit\\">Добавить задачу</button></form></div><div class=\\"col-12\\"><div class=\\"row\\"><div class=\\"col-3\\"><div class=\\"card m-2 text-white bg-primary\\"><div class=\\"card-header\\"><h3 class=\\"card-title\\">Second Task!!</h3></div><div class=\\"card-body\\"><p class=\\"card-text\\">This is my seconds task!</p></div><div class=\\"card-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-light\\">Завершить задачу</button></div></div></div><div class=\\"col-3\\"><div class=\\"card m-2 text-white bg-primary\\"><div class=\\"card-header\\"><h3 class=\\"card-title\\">Hello, World!</h3></div><div class=\\"card-body\\"><p class=\\"card-text\\">This is my first task!</p></div><div class=\\"card-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-light\\">Завершить задачу</button></div></div></div></div></div></div></div>"`;

exports[`TaskContainer - should work finishing task 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Заголовок задачи...\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Текст задачи...\\" value=\\"\\"></div><button class=\\"btn btn-outline-secondary\\" type=\\"submit\\">Добавить задачу</button></form></div><div class=\\"col-12\\"><div class=\\"row\\"><div class=\\"col-3\\"><div class=\\"card m-2 text-white bg-danger\\"><div class=\\"card-header\\"><h3 class=\\"card-title\\">Second Task!!</h3></div><div class=\\"card-body\\"><p class=\\"card-text\\">This is my seconds task!</p></div><div class=\\"card-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-light\\">Удалить задачу</button></div></div></div><div class=\\"col-3\\"><div class=\\"card m-2 text-white bg-primary\\"><div class=\\"card-header\\"><h3 class=\\"card-title\\">Hello, World!</h3></div><div class=\\"card-body\\"><p class=\\"card-text\\">This is my first task!</p></div><div class=\\"card-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-light\\">Завершить задачу</button></div></div></div></div></div></div></div>"`;

exports[`TaskContainer - should work finishing task 2`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Заголовок задачи...\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Текст задачи...\\" value=\\"\\"></div><button class=\\"btn btn-outline-secondary\\" type=\\"submit\\">Добавить задачу</button></form></div><div class=\\"col-12\\"><div class=\\"row\\"><div class=\\"col-3\\"><div class=\\"card m-2 text-white bg-danger\\"><div class=\\"card-header\\"><h3 class=\\"card-title\\">Second Task!!</h3></div><div class=\\"card-body\\"><p class=\\"card-text\\">This is my seconds task!</p></div><div class=\\"card-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-light\\">Удалить задачу</button></div></div></div><div class=\\"col-3\\"><div class=\\"card m-2 text-white bg-danger\\"><div class=\\"card-header\\"><h3 class=\\"card-title\\">Hello, World!</h3></div><div class=\\"card-body\\"><p class=\\"card-text\\">This is my first task!</p></div><div class=\\"card-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-light\\">Удалить задачу</button></div></div></div></div></div></div></div>"`;

exports[`TaskContainer - should work removing task 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Заголовок задачи...\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Текст задачи...\\" value=\\"\\"></div><button class=\\"btn btn-outline-secondary\\" type=\\"submit\\">Добавить задачу</button></form></div><div class=\\"col-12\\"><div class=\\"row\\"><div class=\\"col-3\\"><div class=\\"card m-2 text-white bg-primary\\"><div class=\\"card-header\\"><h3 class=\\"card-title\\">Hello, World!</h3></div><div class=\\"card-body\\"><p class=\\"card-text\\">This is my first task!</p></div><div class=\\"card-footer\\"><button type=\\"button\\" class=\\"btn btn-outline-light\\">Завершить задачу</button></div></div></div></div></div></div></div>"`;

exports[`TaskContainer - should work removing task 2`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-12\\"><form><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Заголовок задачи...\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Текст задачи...\\" value=\\"\\"></div><button class=\\"btn btn-outline-secondary\\" type=\\"submit\\">Добавить задачу</button></form></div><div class=\\"col-12\\"><div class=\\"row\\"></div></div></div></div>"`;
